ext/bcmath/libbcmath/src/recmul.c

```diff
@@ -33,21 +33,32 @@
 #include <stddef.h>
 #include <assert.h>
 #include <stdbool.h>
+#include <limits.h>
 #include "private.h" /* For _bc_rm_leading_zeros() */
 #include "zend_alloc.h"
 
 
 #if SIZEOF_SIZE_T >= 8
 #  define BC_MUL_UINT_DIGITS 8
-#  define BC_MUL_UINT_OVERFLOW 100000000
+#  define BC_MUL_UINT_OVERFLOW (BC_UINT_T) 100000000
+#  define BC_MUL_MAX_ADD_COUNT (ULONG_MAX / (BC_MUL_UINT_OVERFLOW * BC_MUL_UINT_OVERFLOW))
```

Comment:
Note that the size of a long is not necessarily the size of size_t.
I think I prefer a single definition:
`#define BC_MUL_MAX_ADD_COUNT (~((BC_UINT_T) 0) / (BC_MUL_UINT_OVERFLOW * BC_MUL_UINT_OVERFLOW))`

Reply:
It's cool! thx